core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -57,25 +60,46 @@ private TableProperties() {
   public static final String MANIFEST_MERGE_ENABLED = "commit.manifest-merge.enabled";
   public static final boolean MANIFEST_MERGE_ENABLED_DEFAULT = true;
 
-  public static final String DEFAULT_FILE_FORMAT = "write.format.default";
+  private static final String DEFAULT_FILE_FORMAT_CONF = "format.default";
+
+  public static final String DEFAULT_FILE_FORMAT = WRITE_PREFIX + DEFAULT_FILE_FORMAT_CONF;
+  public static final String DELETE_DEFAULT_FILE_FORMAT = DELETE_PREFIX + DEFAULT_FILE_FORMAT_CONF;
   public static final String DEFAULT_FILE_FORMAT_DEFAULT = "parquet";
 
```

Comment:
missing `DELETE_DEFAULT_FILE_FORMAT_DEFAULT`.

I am thinking if we should use avro as the default delete file format instead, because it's a row-based format, and it is likely going to be compacted and removed very soon, so it's not worth the effort for writing deletes in Parquet format. I haven't done any rigorous benchmarking on this, so this is just based on my feeling. Any thoughts?

Reply:
The idea I am following here is to default values for delete format based on data format unless the user sets an explicit value. I think that makes sense for configs like the row group size. I am planning to default the delete file size to some percentage of the target data file.

When it comes to the default delete file format, Avro seems a good candidate on paper as it won't have the extra memory pressure but I'd say the user should opt in in that behavior. We still plan to use min/max stats for filtering and other columnar file format benefits still may be helpful. Right now, I am inclined towards defaulting the delete file format to the default data file format (with the smaller target file size).

I am open here, though. Just my current thinking.